data_pipelines/rne/database/process_rne.py

```diff
@@ -0,0 +1,294 @@
+            code_pays TEXT,
+            code_postal TEXT,
+            commune TEXT,
+            code_insee_commune TEXT,
+            voie TEXT,
+            file_name TEXT
+        )
+    """
+    )
+    cursor.execute(
+        """
+        CREATE TABLE IF NOT EXISTS dirigeants_pm
+        (
+            siren TEXT,
+            date_mise_a_jour TEXT,
+            denomination TEXT,
+            siren_dirigeant TEXT,
+            role TEXT,
+            forme_juridique TEXT,
+            pays TEXT,
+            code_pays TEXT,
+            code_postal TEXT,
+            commune TEXT,
+            code_insee_commune TEXT,
+            voie TEXT,
+            file_name TEXT
+        )
+    """
+    )
+    create_index_db(cursor)
+
+
+def create_index_db(cursor):
+    index_statements = [
+        "CREATE INDEX IF NOT EXISTS idx_siren_pp ON dirigeants_pp (siren);",
+        "CREATE INDEX IF NOT EXISTS idx_siren_pm ON dirigeants_pm (siren);",
+        "CREATE INDEX IF NOT EXISTS file_pp ON dirigeants_pp (file_name);",
+        "CREATE INDEX IF NOT EXISTS file_pm ON dirigeants_pm (file_name);",
+        """CREATE INDEX IF NOT EXISTS idx_pp_siren_file_name
+        ON dirigeants_pp (siren, file_name);""",
+        """CREATE INDEX IF NOT EXISTS idx_pm_siren_file_name
+        ON dirigeants_pm (siren, file_name);""",
+    ]
+
+    for statement in index_statements:
+        cursor.execute(statement)
+
+
+def inject_records_into_db(file_path, db_path, file_type):
+    dirigeants_pp, dirigeants_pm = [], []
+
+    with open(file_path, "r") as file:
+        logging.info(f"Processing stock file: {file_path}")
+        try:
+            if file_type == "stock":
+                json_data = file.read()
+                data = json.loads(json_data)
+                for record in data:
+                    list_dirigeants_pp, list_dirigeants_pm = extract_dirigeants_data(
+                        record, file_type
+                    )
+                    dirigeants_pp = dirigeants_pp + list_dirigeants_pp
+                    dirigeants_pm = dirigeants_pm + list_dirigeants_pm
+            elif file_type == "flux":
+                for line in file:
+                    data = json.loads(line)
+                    for record in data:
+                        (
+                            list_dirigeants_pp,
+                            list_dirigeants_pm,
+                        ) = extract_dirigeants_data(record, file_type)
+                        dirigeants_pp = dirigeants_pp + list_dirigeants_pp
+                        dirigeants_pm = dirigeants_pm + list_dirigeants_pm
+        except json.JSONDecodeError as e:
+            raise Exception(f"JSONDecodeError: {e} in file {file_path}")
+        insert_dirigeants_into_db(dirigeants_pp, dirigeants_pm, file_path, db_path)
+
+
+def find_and_delete_same_siren(cursor, siren, file_path):
+    """
+    Find and delete rows with the same SIREN and a different file
+    name from the database.
```

Comment:
A bit unclear as of why you delete the rows

Reply:
Is it absolute duplicate ? Or is there a strategy to chose which row to keep / delete

Reply:
No, I delete the rows because the data had been modified and I want to keep the latest data returned by RNE. Especially when adding flux files to stock files already inserted the database.
Say `siren a` with `file_1`, exists in the db but this siren has been modified in `file_2`, I want to delete the records with `file_1` and only keep the ones from `file_2`.

Reply:
Ok I suspected this was the reason and I think it can be more obvious in the code and the comments. 

For instance : 
`# If existing rows are found, delete them`

-> `# If existing rows are found, delete them as they are outdated`

Reply:
Noted!